sympy/solvers/solveset.py

```diff
@@ -76,6 +76,14 @@ def _invert_real(f, g_ys, symbol):
                             imageset(Lambda(n, f.inverse()(n)), g_ys), symbol)
 
     if isinstance(f, Abs):
+        _a = []
+        for i in g_ys:
+            try:
+                if i < 0:
+                    _a.append(i)
+            except TypeError:
+                pass
+        g_ys = g_ys - FiniteSet(*_a)
```

Comment:
Probably i regret about making such a silly mistake, i could surely make this better.